### ISSUE TITLE
Wallet empty state has disabled "Add bank account" button for new, unvalidated accounts

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#48041]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.25-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1724687808309139

## Action Performed:
1. Sign up a new public domain account
2. Click on avatar > Wallet
3. Click on enable wallet
## Expected Result:
Step 2 and 3"Add bank account" should be enabled

## Actual Result:
"Add bank account" is disabled and grayed out
For the new unvalidated account, we'd mimic something like this where we just ask you to verify your account:
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![CleanShot 2024-08-26 at 11 57 03@2x](https://github.com/user-attachments/assets/f0b86367-80d6-4f81-a2f1-c38cbdead737)
![CleanShot 2024-08-26 at 11 55 25@2x](https://github.com/user-attachments/assets/1e46c38f-645b-4792-b15f-16bf0dc2eede)
![CleanShot 2024-08-26 at 11 57 40@2x](https://github.com/user-attachments/assets/c24a76e7-b0ef-4eee-ad49-638b0f3e5234)
![Snip - (2) New Expensify - Google Chrome (3)](https://github.com/user-attachments/assets/601cd38f-42ba-40ae-87ff-fcb6add6a688)
![Snip - (2) New Expensify - Google Chrome (2)](https://github.com/user-attachments/assets/41c5bac9-624c-4c4b-858e-b9ec8dece62d)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by shawnborton** [comment]
@Krishna2323 rather than throwing an error modal, I was thinking we could show the validate UI in the RHP.

**Comment by shawnborton** [comment]
Actually looks like @cretadn22 already proposed that.

**Comment by shawnborton** [comment]
I think it would be simpler if we just added some text in that RHP view that explains why we need to validate for that particular feature.

**Comment by shawnborton** [comment]
I think we should do something simple like this, where we basically have a little line that says "This feature requires you to validate your account" and then we show the validate form, something like this:
![CleanShot 2024-08-27 at 12 55 19@2x](https://github.com/user-attachments/assets/307d8f22-1835-4f55-85b7-629d7df7e351)


cc @Expensify/design in case you have any additional thoughts.

**Comment by dannymcclain** [comment]
I like what you're proposing Shawn. Keeps you in the flow and doesn't derail you, but still gets you to validate.

**Comment by trjExpensify** [comment]
Samesies! ^^

**Comment by dubielzyk-expensify** [comment]
Dig it. Would it be a bit clearer if we have the header say "Validate your account"?

**Comment by shawnborton** [comment]
@dubielzyk-expensify yeah totally agree with that, I just grabbed a quick example from the contact methods flow since that is the only spot (I think?) that has this kind of in-product account validation thus far.

**Comment by miljakljajic** [comment]
@shawnborton whose proposal are we going with here? I'll get them assigned

**Comment by shawnborton** [comment]
I think that would be @jjcoffee's decision right?

**Comment by jjcoffee** [comment]
Hmm I'm not sure if I agree with navigating to the contact details page here. We wouldn't normally reuse a _page_ for a different purpose, but rather the components in the page (`ValidateCodeForm` in this case). I think navigating to a completely different route also will cause issues; if you refresh after tapping `Add bank account`, for example, you'd lose your place (the central pane would switch in the background to the profile screen).

I think I'd be more in favour of adding a `ValidateCodeForm` either in a new page or maybe as the first step of the add bank account flow (only if you aren't validated of course!).

**Comment by jjcoffee** [comment]
@daledah In that case it makes sense since the purpose is the same (adding an email address) and so we're directing the user to the page appropriate to that purpose. 

Once we start talking about customising the _contact details_ page to add text and a header that's specific to this flow, I think it doesn't really make sense to reuse the page anymore.

**Comment by jjcoffee** [comment]
Thanks @daledah, happy to go with your updated [proposal](https://github.com/Expensify/App/issues/48041#issuecomment-2311388051)!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by jjcoffee** [comment]
@daledah Sure, asked [on Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1726047320703039). Are you waiting to be added to Slack? Maybe @miljakljajic can help get you added if so.

**Comment by jjcoffee** [comment]
@daledah Here you go:

EN: `This feature requires you to validate your account.`
ES: `Esta función requiere que valides tu cuenta.`

EN: `Validate your account.`
ES: `Valida tu cuenta.`

**Comment by c3024** [comment]
Hi, we have a `ValidateContactActionPage` added here #48320. This is to ask for validating the primary contact every time when trying to add a secondary contact. I think the page being added in this issue is also similar, though it is meant only if the primary contact is not validated. There are few flows which need primary contact verification every time here #48541.

Is it better to use this `ValidateContactActionPage` and make it a common page for all these flows that require validation of the primary contact before allowing an action?

**Comment by jjcoffee** [comment]
@c3024 Oh interesting, thanks for bringing that to our attention! 

I wonder if we could be better off using `ValidateCodeActionModal` (added in https://github.com/Expensify/App/pull/48628) for our purposes here, as we are not adding a step in the process of adding a bank account but a one-off block to verify your account (so a modal makes more sense in my mind). I don't know if that makes a big difference to the implementation @daledah?

**Comment by miljakljajic** [comment]
Apologies @daledah we've run out of single channel guest spaces in slack - are you still waiting to be added? I'll make sure you're at the top of the list

**Comment by jjcoffee** [comment]
@daledah Yes good point, we do want to handle it in both places.

**Comment by miljakljajic** [comment]
> @daledah  Hey yes, I've sent multiple emails to contributors@expensify.com on the matter but haven't received responses recently, would be great if you could help.

Unfortunately we've run out of spaces in the slack workspace, but I've added you to the top of the list for when the next batch frees up

**Comment by ikevin127** [comment]
❓ **Question regarding the new verification screen in RHP (`ROUTES.SETTINGS_WALLET_VERIFY_ACCOUNT`):**
Is it expected behaviour to be logged out once you introduced the magic code ?

ℹ️ For context, I'm asking because we held issue https://github.com/Expensify/App/issues/49523 until the PR for this one was merged, I observed this behaviour while testing our PR and wanted to confirm that being logged out is the correct behaviour.

cc @arosiclair @jjcoffee @daledah

**Comment by ikevin127** [comment]
Yes, the user is logged out instead of opening the Add bank account page, here's a video:

https://github.com/user-attachments/assets/06ab7cb5-74bf-4f13-9d3e-238450c24d62

I'm trying to figure out if this is a regression of this issue's implementation or we didn't use your implementation correctly and the regression is on our [issue](https://github.com/Expensify/App/issues/49523)'s side.

From what I can see, your PR is using the route with redirect to add BA:

`Navigation.navigate(ROUTES.SETTINGS_WALLET_VERIFY_ACCOUNT.getRoute(ROUTES.SETTINGS_ADD_BANK_ACCOUNT));`

whereas in our PR, the author did not pass anything:

`Navigation.navigate(ROUTES.SETTINGS_WALLET_VERIFY_ACCOUNT.getRoute());`

which might be the cause of the logout issue 🤔

**Comment by ikevin127** [comment]
@daledah Thanks for looking into the root cause! 🚀

Are you experiencing the logout upon validation as well in this issue's flow testing ?

##

This is definitely weird, I mean it makes sense that when you validate an unvalidated account the `authToken` would change, but because a call is made right after with the outdated key -> the user is logged out (expected if you ask me).

Given this behaviour I'm leaning towards the scenario where none of the issues (this or ours) is the direct cause of the logout, but rather because the logic we implemented, we revealed that the FE logic wasn't designed to account for the edge case of transitioning between unvalidated to validated account and changing `authToken`.

##

In this case I'm not sure who should handle a fix for this edge case in a follow-up 🤔

Tagging the assigned CME's of both issues for visibility ⏬

cc @arosiclair from this issue
cc @nkuoch from issue https://github.com/Expensify/App/issues/49523

**Comment by arosiclair** [comment]
Thanks for looking into the issue @daledah @ikevin127. Is this reproducible in staging or `main` with any other flow at the moment? If so, we can report the bug and start investigating in another issue.

**Comment by arosiclair** [comment]
Nevermind, looks like it is [here](https://github.com/Expensify/App/issues/50284). It sounded like we did not merge the PR for https://github.com/Expensify/App/issues/49523 yet, but it looks like we did? I think we probably need to revert it and find a fix for the root problem.

**Comment by arosiclair** [comment]
> Read through the above and it sounds like I can still pay this as normal on 10-11 correct?

Yeah that's correct.

**Comment by strepanier03** [comment]
### Payment Summary

- [x] $250 - @jjcoffee requires payment paid via Upwork
- [ ] $250 - @daledah via New Expensify request, feel free to put in that request now.

**Comment by strepanier03** [comment]
Just sent you the offer and msg'd in Slack. Will pay out when I get confirmation it's accepted.
